### ISSUE TITLE
Why clamping the grouping columns ? 

### ISSUE THREAD

**Issue by vserret** [opendp/smartnoise-sdk#359]
The tau-thresholding makes sure that releasing groups is safe. So I think there is no use to clamp the grouping columns [here](https://github.com/opendp/smartnoise-sdk/blob/85dd0101af30fff593fdad2a8193868aaceedc37/sdk/opendp/smartnoise/sql/private_rewriter.py#L250). 

We could replace: 
```
select = [
                self.clamp_expression(ne, relations, child_scope, self.options.clamp_columns)
                for ne in query.select.namedExpressions
            ]
```
by: 
```
select = [
                self.clamp_expression(ne, relations, child_scope, 
               (self.options.clamp_columns is_grouping_expression(ne))
                for ne in query.select.namedExpressions
            ]
```
where we have to define `is_grouping_expression`. 
By doing that, we should be able to allow grouping by expressions with minimal changes.

**Comment by joshua-oss** [comment]
That's an interesting observation.  My assumption was that grouping columns will typically be categorical, so won't be clamped.  But I could imagine someone (for example) doing `SELECT Salary, SUM(Bonus) GROUP BY Salary`, where `Salary` has a `minval` and `maxval`.  Is this what you were thinking of?  Or do you mean when grouping by some expression like `GROUP BY SomeVal % 5`?

In any case, I agree the grouping column needn't be clamped.

**Comment by joshua-oss** [comment]
Issue fixed in #360
